website/project/views/contributor.py

```diff
@@ -822,6 +823,18 @@ def claim_user_form(auth, **kwargs):
             if provider:
                 redirect_url = web_url_for('auth_login', next=provider.landing_url, _absolute=True)
             else:
+                node = AbstractNode.load(pid)
+                preprint = Preprint.load(pid)
```

Comment:
So `pid` in this view is a unique key for the object that used to just be a node but now could be a node or a preprint?

Reply:
So prior to this change, PID could be a node or a preprint. What this the change does is capturing the PID and see what node/preprint it is and assign system tags to users accordingly.